detectable/src/main/java/com/synopsys/integration/detectable/detectables/setuptools/transform/SetupToolsGraphTransformer.java

```diff
@@ -0,0 +1,152 @@
+        this.executableRunner = executableRunner;
+    }
+
+    public DependencyGraph transform(ExecutableTarget pipExe, SetupToolsParsedResult parsedResult) throws ExecutableRunnerException {
+        DependencyGraph dependencyGraph = new BasicDependencyGraph();
+        
+        if (pipExe != null) {
+            // Get dependencies by running pip show on each direct dependency
+            for (PythonDependency directDependency : parsedResult.getDirectDependencies()) {
+                handleShowDependency(pipExe, dependencyGraph, directDependency.getName(), null, directDependency.isConditional());
+            }
+        } else {
+            // Unable to determine transitive dependencies, add parsed dependencies directly
+            // to the root of the graph.
+            handleParsedDependencies(parsedResult, dependencyGraph);
+        }
+        
+        return dependencyGraph;
+    }
+
+    private void handleParsedDependencies(SetupToolsParsedResult parsedResult, DependencyGraph dependencyGraph) {
+        List<PythonDependency> directDependencies = parsedResult.getDirectDependencies();
+        
+        for (PythonDependency directDependency : directDependencies) {
+            String name = directDependency.getName();
+            String version = directDependency.getVersion();
+            
+            Dependency currentDependency;
+            if (StringUtils.isEmpty(version)) {
+                currentDependency = entryToDependency(name);
+            } else {
+                currentDependency = entryToDependency(name, version);
+            }
+            dependencyGraph.addChildrenToRoot(currentDependency);
+        }
+    }
+    
+    private void handleShowDependency(ExecutableTarget pipExe, DependencyGraph dependencyGraph, String dependencyToSearch, Dependency parentDependency, boolean isConditionalDependency) throws ExecutableRunnerException {
+        List<String> rawShowOutput = runPipShow(pipExe, dependencyToSearch, isConditionalDependency);
+        
+        if (rawShowOutput.isEmpty()) {
+            return;
+        }
+        
+        Map<String, String> showOutput = parsePipShow(rawShowOutput);
+        
+        Dependency currentDependency = entryToDependency(dependencyToSearch, showOutput.get("Version"));
+        
+        if (parentDependency == null) {
+            // No parent, add to root of graph
+            dependencyGraph.addChildrenToRoot(currentDependency);
+        } else {
+            dependencyGraph.addChildWithParent(currentDependency, parentDependency);
+        }
+        
+        // See if we need to continue exploring this chain of dependencies
+        if (showOutput.containsKey(REQUIRES)) {
+            String[] requiredPackages = showOutput.get(REQUIRES).split(", ");
+            for (String requiredPackage : requiredPackages) {
+                handleShowDependency(pipExe, dependencyGraph, requiredPackage, currentDependency, false);
```

Comment:
Is it possible that requiredPackages returns an empty string or is null? If so, it might mess up the runPipShow command and throw an exception. Ideally we want to skip that package in that case correct?

Reply:
I don't believe this is possible. pip show returns the Requires line regardless of it anything is required or not. In cases where there is nothing required it will return
Requires:
instead of something like
Requires: package-name

In the case of Requires with nothing after it but a : the split will only contain one part (see method parsePipShow in this file) and then we won't add the Requires key to showOutput so line 98 above will never trigger. It should only trigger when we have at least one requires.